src/rpft/parsers/creation/flowparser.py

```diff
@@ -497,6 +512,21 @@ def _get_or_create_group(self, name, uuid=None):
             uuid = None
         return Group(name=name, uuid=uuid)
 
+    def _validate_webhook_headers(self, headers):
+        # Dict is not yet supported in the row parser,
+        # so we need to convert a list of pairs into dict.
+        if type(headers) == dict:
+            return headers
+        elif type(headers) == list:
+            if headers == [""]:
+                # Future row parser should return [] instead of [""]
+                return {}
+            if not all(map(lambda x: type(x) == list and len(x) == 2, headers)):
+                LOGGER.critical(f"Webhook headers must be a list of pairs.")
```

Comment:
My understanding is that logging at the 'critical' level will cause the process to exit. I find this behaviour unusual and surprising for a logger, and I would recommend raising exceptions instead.

I don't expect anything to be done about this now because it's likely to be quite a substantial change, but it would be helpful to try to deprecate the use of `LOGGER.critical` to exit the program.

Reply:
The reason for using `LOGGER.critical` rather than Exceptions is that the logger has context information on which particular row in which particular sheet the error is in, and thus errors in the data can be traced down much more easily by the user. This wouldn't be possible with plain exception, except if I passed down this information in every single function call.

Reply:
My main concern is about the shutdown behaviour, but I can see that this only happens when the toolkit is used as a CLI command and not when it is used as a library. I think this deserves to be discussed further, but I will raise a new issue for this.